packages/offix-datastore/src/Model.ts

```diff
@@ -171,15 +151,19 @@ export class Model<T = unknown> {
     if (data.data && data.data.length > 0) {
       data.data
         .filter((d: any) => (d._deleted))
-        .forEach((d: any) => { this.remove(matcher(d)); });
+        .forEach((d: any) => {
```

Comment:
This code will be removed - DeltaSync should not happen in the model. 
The model will just expose it's own replication proxy